### PR TITLE
Feature/add validator to pixel routes

### PR DESCRIPTION
**Pull request by aweolumidedavid**
- [x] issue #5
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

Add checks to all the pixel routes to validation position and color ( make sure they are in the correct range ) before using them. The following route functions were modified:
- consumeIndexerMsg
- getPixel
- placePixelDevnet
- placePixelRedis

### CODE REVIEW

File: backend/routes/indexer.go [hunk 1]
```diff
@@ -58,86 +58,145 @@ func InitIndexerRoutes() {
...
 		return
 	}
 
-	address := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[1]
-	address = address.(string)[2:]
-	posHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[2]
-	dayIdxHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["keys"].([]interface{})[3]
-	colorHex := reqBody["data"].(map[string]interface{})["batch"].([]interface{})[0].(map[string]interface{})["events"].([]interface{})[0].(map[string]interface{})["event"].(map[string]interface{})["data"].([]interface{})[0]
+	// Extract necessary fields from the request body
+	data, ok := reqBody["data"].(map[string]interface{})
```
Comment (b-j-roberts) [inline backend/routes/indexer.go:79]: Sorry, but there are other plans to extract this data differently. If you could revert these changes and just do the pixels color and position validation, that would be prefered.

File: backend/routes/indexer.go [hunk 2]
```diff
@@ -58,86 +58,145 @@ func InitIndexerRoutes() {
...
+		return
+	}
+
+	// Validate position and color
+	if position < 0 || position >= int64(core.ArtPeaceBackend.CanvasConfig.Canvas.Width*core.ArtPeaceBackend.CanvasConfig.Canvas.Height) {
+		handleError(w, "Invalid position value", http.StatusBadRequest)
+		return
+	}
+
+	if color < 0 || color >= (1 << core.ArtPeaceBackend.CanvasConfig.ColorsBitWidth) {
```
Comment (b-j-roberts) [inline backend/routes/indexer.go:139]: Checking against `core.ArtPeaceBackend.CanvasConfig.Colors` length would be better here.

File: backend/routes/pixel.go [hunk 3]
```diff
@@ -57,60 +65,108 @@ func getPixelInfo(w http.ResponseWriter, r *http.Request) {
 }
 
 func placePixelDevnet(w http.ResponseWriter, r *http.Request) {
+    reqBody, err := io.ReadAll(r.Body)
```
Comment (b-j-roberts) [inline backend/routes/pixel.go:68]: This should be after the production check

File: backend/routes/pixel.go [hunk 4]
```diff
@@ -57,60 +65,108 @@ func getPixelInfo(w http.ResponseWriter, r *http.Request) {
...
+
+    // Validate position range
+    if position < 0 || position >= int(core.ArtPeaceBackend.CanvasConfig.Canvas.Width*core.ArtPeaceBackend.CanvasConfig.Canvas.Height) {
+        http.Error(w, "Position out of range", http.StatusBadRequest)
+        return
+    }
+
+    // Validate color format (e.g., validate against allowed colors)
+    // Example: check if color is a valid hex color code
+    if !isValidHexColor(color) {
```
Comment (b-j-roberts) [inline backend/routes/pixel.go:102]: This color is a color index, so the checks should be the same as the others.

File: backend/routes/pixel.go [hunk 5]
```diff
@@ -57,60 +65,108 @@ func getPixelInfo(w http.ResponseWriter, r *http.Request) {
...
+    canvasHeight := core.ArtPeaceBackend.CanvasConfig.Canvas.Height
+
+    // Validate position range
+    if position >= canvasWidth*canvasHeight {
+        http.Error(w, "Position out of range", http.StatusBadRequest)
+        return
+    }
+
+    // Validate color range (e.g., ensure color value fits within bit width)
+    if color >= (1 << core.ArtPeaceBackend.CanvasConfig.ColorsBitWidth) {
```
Comment (b-j-roberts) [inline backend/routes/pixel.go:149]: `< 0` check and colors length like other comment

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: backend/routes/indexer.go (hunk 1 above)
Comment (aweolumidedavid) [reply backend/routes/indexer.go:79]: Noted, I will revert back

File: backend/routes/indexer.go (hunk 2 above)
Comment (aweolumidedavid) [reply backend/routes/indexer.go:139]: core.ArtPeaceBackend.CanvasConfig.Colors.length, Noted.

File: backend/routes/pixel.go (hunk 3 above)
Comment (aweolumidedavid) [reply backend/routes/pixel.go:68]: Oh, I understand. Noted

File: backend/routes/indexer.go (hunk 1 above)
Comment (b-j-roberts) [reply backend/routes/indexer.go:79]: Thank you!

**Review by b-j-roberts — APPROVED** [review]
Thank you!
